openslide/deepzoom.py

```diff
@@ -79,13 +86,13 @@ def __init__(self, osr, tile_size=254, overlap=1, limit_bounds=False):
                 for prop, l0_lim in zip(self.BOUNDS_SIZE_PROPS, osr.dimensions)
             )
             # Dimensions of active area
-            self._l_dimensions = tuple(
-                tuple(
-                    int(math.ceil(l_lim * scale))
-                    for l_lim, scale in zip(l_size, size_scale)
+            self._l_dimensions = [
+                (
+                    int(math.ceil(l_size[0] * size_scale[0])),
+                    int(math.ceil(l_size[1] * size_scale[1])),
```

Comment:
I've changed the zip pattern in a few places to directly construct the 2-element tuples so that the type checker knows it's a 2-element tuple.